server/src/test/java/org/elasticsearch/action/search/KnnSearchSingleNodeTests.java

```diff
@@ -383,7 +383,7 @@ public void testKnnFilteredAlias() throws IOException {
         indicesAdmin().prepareRefresh("index").get();
 
         float[] queryVector = randomVector();
-        KnnSearchBuilder knnSearch = new KnnSearchBuilder("vector", queryVector, 10, 50, null, null);
+        KnnSearchBuilder knnSearch = new KnnSearchBuilder("vector", queryVector, 10, 50, 10f, null, null);
```

Comment:
I'm looking at these examples and I think those should be an IAE - they're specifying both `num_candidates` and `visitPercentage`.

Reply:
We should allow users to specify both, especially if they are querying multiple indices. 

We should also allow only specifying one, even if the backing indices is the completely wrong type (e.g. visit_percentage, but only querying HNSW, or only num_candidates and only querying diskbbq). 

Throwing an error based on your index pattern being queried is bad.

Reply:
> We should allow users to specify both, especially if they are querying multiple indices.

Makes sense, thanks!